### PR TITLE
[interpreter] Dune test suite

### PR DESCRIPTION
**Pull request by vapourismo**
This PR adds a few tweaks to allow us to run the test suite against the reference interpreter with a simple call to `dune test`.

### CODE REVIEW

File: interpreter/dune [hunk 1]
```diff
@@ -6,7 +6,8 @@
  ; WASM REPL every time in all the dependencies.
```
Comment (rossberg) [inline interpreter/dune:6]:
```suggestion
 ; Wasm REPL every time in all the dependencies.
```

**Review by rossberg — APPROVED** [review]
Looks good to me. Would it be possible to have wrapper targets in the Makefile that invoke these?

**Comment by vapourismo** [comment]
> Would it be possible to have wrapper targets in the Makefile that invoke these?

Sure. Like `make dune-test` for instance?

File: interpreter/Makefile [hunk 2]
```diff
@@ -136,7 +136,7 @@ TESTDIR =	../test/core
 TESTFILES =	$(shell cd $(TESTDIR); ls *.wast; ls [a-z]*/*.wast)
 TESTS =		$(TESTFILES:%.wast=%)
 
-.PHONY:		test debugtest partest
+.PHONY:		test debugtest partest dune-test
```
Comment (rossberg) [inline interpreter/Makefile:139]: Nit: `dunetest` for consistency.

Can you also add such a target for the dune build itself?

**Review by rossberg — APPROVED** [review]

File: interpreter/Makefile (hunk 2 above)
Comment (vapourismo) [reply interpreter/Makefile:141]: Fixed and added.

**Review by rossberg — APPROVED** [review]
Thanks!

**Pull request merged**